python/coinbase-agentkit/coinbase_agentkit/wallet_providers/cdp_evm_server_wallet_provider.py

```diff
@@ -0,0 +1,344 @@
+
+    def send_transaction(self, transaction: TxParams) -> HexStr:
+        """Send a transaction to the network.
+
+        Args:
+            transaction (TxParams): Transaction parameters including to, value, and data
+
+        Returns:
+            HexStr: The transaction hash as a hex string
+
+        """
+        client = self.get_client()
+
+        async def _send_transaction():
+            async with client as cdp:
+                return await cdp.evm.send_transaction(
+                    address=self.get_address(),
+                    transaction=TransactionRequestEIP1559(
+                        to=transaction["to"],
+                        value=transaction.get("value", 0),
+                        data=transaction.get("data", "0x"),
+                    ),
+                    network=self._network.network_id,
+                )
+
+        return self._run_async(_send_transaction())
+
+    def wait_for_transaction_receipt(
+        self, tx_hash: HexStr, timeout: float = 120, poll_latency: float = 0.1
+    ) -> dict[str, Any]:
+        """Wait for transaction confirmation and return receipt.
+
+        Args:
+            tx_hash (HexStr): The transaction hash to wait for
+            timeout (float): Maximum time to wait in seconds, defaults to 120
+            poll_latency (float): Time between polling attempts in seconds, defaults to 0.1
+
+        Returns:
+            dict[str, Any]: The transaction receipt as a dictionary
+
+        Raises:
+            TimeoutError: If transaction is not mined within timeout period
+
+        """
+        return self._web3.eth.wait_for_transaction_receipt(
+            tx_hash, timeout=timeout, poll_latency=poll_latency
+        )
+
+    def sign_message(self, message: str | bytes) -> HexStr:
+        """Sign a message using the wallet's private key.
+
+        Args:
+            message (str | bytes): The message to sign, either as a string or bytes
+
+        Returns:
+            HexStr: The signature as a hex string
+
+        """
+        client = self.get_client()
+
+        async def _sign_message():
+            async with client as cdp:
+                return await cdp.evm.sign_message(
+                    address=self.get_address(),
+                    message=message,
+                )
+
+        return self._run_async(_sign_message())
+
+    def sign_typed_data(self, typed_data: dict[str, Any]) -> HexStr:
+        """Sign typed data according to EIP-712 standard.
+
+        Args:
+            typed_data (dict[str, Any]): The typed data to sign following EIP-712 format
+
+        Returns:
+            HexStr: The signature as a hex string
+
+        """
+        client = self.get_client()
+
+        async def _sign_typed_data():
+            async with client as cdp:
+                return await cdp.evm.sign_typed_data(
+                    address=self.get_address(),
+                    typed_data=typed_data,
+                )
```

Comment:
There is no top-level `typed_data` parameter. See here for a usage: https://github.com/coinbase/cdp-sdk/blob/main/examples/python/evm/account.sign_typed_data.py